File: VowelsOfAllSubstrings.cpp
class Solution {
public:
    bool isVowel(char c)
    {
        if(c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u')
            return true;
        else
            return false;
    }
    long long countVowels(string word) {
        
        long long sum = 0;
        long long n = word.size();
        
        for(int i = 0 ; i < n ; i++)
        {
            if(isVowel(word[i]))
            {
                if(i == 0 || i == n - 1)
                    sum += n;
                else
                  sum += ((n + (i * (n - (i + 1)))));  
            }
        }
        

        
        return sum;
    }
};

File: VowelsofAllSubstrings.py
    def isVowel(self, c: str) -> bool:
        if c in ['a', 'e', 'i', 'o', 'u']:
            return True
        else:
            return False

    def countVowels(self, word: str) -> int:
        sum = 0
        n = len(word)

        for i in range(n):
            if self.isVowel(word[i]):
                sum += (n - i) * (i + 1)

        return sum
